src/H5Tconv_float.c

```diff
@@ -42,54 +44,34 @@ H5T__conv_f_f(const H5T_t *src_p, const H5T_t *dst_p, H5T_cdata_t *cdata, const
               size_t nelmts, size_t buf_stride, size_t H5_ATTR_UNUSED bkg_stride, void *buf,
               void H5_ATTR_UNUSED *bkg)
 {
-    /* Traversal-related variables */
```

Comment:
The entirety of the conversion logic here was moved into a new `H5T__conv_f_f_loop` helper function so it can be reused when converting between complex number types. No changes to the existing actual conversion logic were made, but some additions were made and refactoring done for readability and reusability. Most notably, the logic that checks for special floating-point values (+/- Inf, NaN, 0) was refactored into a helper function `H5T__conv_float_find_special` that returns an enum value indicating the type of special value found, if any. This compacted the logic a bit and made it reusable since both parts of a complex number have to be checked and the logic would have to have been duplicated otherwise. A few small additions were made just to do the right thing for complex numbers vs. floating-point numbers.